contracts/test/CompoundGovernorSetWhitelistGuardian.t.sol

```diff
@@ -0,0 +1,51 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity 0.8.26;
+
+import {ProposalTest} from "contracts/test/helpers/ProposalTest.sol";
+import {IGovernor} from "@openzeppelin/contracts/governance/IGovernor.sol";
+import {CompoundGovernor} from "contracts/CompoundGovernor.sol";
+
+contract CompoundGovernorSetWhitelistGuardianTest is ProposalTest {
+    function _buildSetWhitelistGuardianProposal(address _whitelistGuardian)
+        private
+        view
+        returns (Proposal memory _proposal)
+    {
+        address[] memory _targets = new address[](1);
+        _targets[0] = address(governor);
+
+        uint256[] memory _values = new uint256[](1);
+        _values[0] = 0;
+
+        bytes[] memory _calldatas = new bytes[](1);
+        _calldatas[0] = abi.encodeWithSelector(CompoundGovernor.setWhitelistGuardian.selector, _whitelistGuardian);
+
+        _proposal = Proposal(_targets, _values, _calldatas, "Set New whitelistGuardian");
+    }
+
+    function testFuzz_SetsWhitelistGuardianAsTimelock(address _whitelistGuardian) public {
+        Proposal memory _proposal = _buildSetWhitelistGuardianProposal(_whitelistGuardian);
+        _submitPassQueueAndExecuteProposal(delegatee, _proposal);
+        assertEq(governor.whitelistGuardian(), _whitelistGuardian);
+    }
+
+    function testFuzz_EmitsEventWhenAWhitelistGuardianIsSet(address _whitelistGuardian) public {
+        Proposal memory _proposal = _buildSetWhitelistGuardianProposal(_whitelistGuardian);
+        _submitPassAndQueProposal(delegatee, _proposal);
+
+        vm.expectEmit();
+        emit CompoundGovernor.WhitelistGuardianSet(governor.whitelistGuardian(), _whitelistGuardian);
+
+        vm.prank(TIMELOCK_ADDRESS);
```

Comment:
call shouldn't have to come from timelock, can you fuzz the caller (and assume that caller isn't proxy admin?)

Reply:
https://github.com/ScopeLift/compound-governance-upgrade/pull/29/commits/084a46b57e2c4f8805dadfbd556ca383b53bed3f

Reply:
Assume it's not proxy admin in https://github.com/ScopeLift/compound-governance-upgrade/pull/29/commits/c3995f33a7e8ae7d6f167ca872cda67d7c9f0814